pkg/controller/cdapmaster/cdapmaster_controller.go

```diff
@@ -370,10 +472,74 @@ func (b *Base) Objects(rsrc interface{}, rsrclabels map[string]string, observed,
 	}
 	var err error
 	for k, v := range configs {
-		resources, err = addConfigMapItem(master, getConfigName(master, k), labels, v, resources)
+		resources, err = addConfigMapItem(master, getResourceName(master, k), labels, v, resources)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	// User has reset the CR after failure. Remove upgrade failed status.
+	if master.Spec.Image == master.Status.ImageToUse && master.Status.Meta.IsConditionTrue(upgradeFailed) {
+		master.Status.ClearCondition(upgradeFailed, upgradeResetMessage, upgradeResetMessage)
+	}
+
+	if !isImageUpdatePending(master) {
```

Comment:
Since there is only one usage of this method, I think it's better inline the logic here directly (with comment), which make it clearer for the condition and intention is.